http_submission/sample_evaluation.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+import pickle
+import requests
+import sys
+from sample_submission import MySimpleScanpathModel
+from PIL import Image
+from io import BytesIO
+import json
+import matplotlib.pyplot as plt
+from pysaliency.plotting import plot_scanpath
+sys.path.insert(0, '..')
+import pysaliency
+
+class HTTPScanpathModel(MySimpleScanpathModel):
+    def __init__(self, url):
+        self.url = url
+        self.log_density_url = url + "/conditional_log_density"
+        self.type_url = url + "/type"
+
+
+
+
+    def conditional_log_density(self, stimulus, x_hist, y_hist, t_hist, attributes=None, out=None):
+
+        # build request
+        pil_image = Image.fromarray(stimulus)
+        image_bytes = BytesIO()
+        pil_image.save(image_bytes, format='png')
+
+        def _convert_attribute(attribute):
+            if isinstance(attribute, np.ndarray):
+                return attribute.tolist()
+            return attribute
+
+        json_data = {
+            "x_hist": list(x_hist),
+            "y_hist": list(y_hist),
+            "t_hist": list(t_hist),
+            "attributes": {key: _convert_attribute(value) for key, value in (attributes or {}).items()}
+        }
+
+        # send request
+        response = requests.post(f"{self.log_density_url}", data={'json_data': json.dumps(json_data)}, files={'stimulus': image_bytes.getvalue()})
+
+        # parse response
+        if response.status_code != 200:
+            raise ValueError(f"Server returned status code {response.status_code}")
+
+        return np.array(response.json()['log_density'])
+
+    def type(self):
```

Comment:
the type mainly needs to be checked to make sure that the server is compatible with this model. So I would suggest to call the function something like `check_type`:
```
def check_type(self):
    response = requests...
    if not response['type'] == 'ScanpathModel':
        raise ...
    if not response['version'] in ['v1.0.0']:
        raise ...
```

Then you can call `check_type` in `__init__` to make sure that the server is valid.